src/inputdevice.cpp

```diff
@@ -5003,30 +5063,45 @@ static int handle_input_event2(int nr, int state, int max, int flags, int extra)
 	int autofire = (flags & HANDLE_IE_FLAG_AUTOFIRE) ? 1 : 0;
 
 	if (nr <= 0 || nr == INPUTEVENT_SPC_CUSTOM_EVENT)
+	{
+		handle_vkbd();
```

Comment:
Same here, perhaps it would be best to add an `#ifdef AMIBERRY` for this, since it goes in code that WinUAE uses, and we want to continue merging updates from it. Having a separate block clearly stating that this is Amiberry-specific makes it easier to merge things an leave our customizations in place. :)

Reply:
I made these changes